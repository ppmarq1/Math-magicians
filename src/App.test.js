import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom/extend-expect';
import Calculator from './components/Calculator';
import Home from './components/home';
import NavBar from './components/navbar';
import Quotes from './components/quote';
import calculate from './logic/calculate';
import operate from './logic/operate';

describe('Functions test', () => {
  test('calculate.js test', () => {
    const initialItems = {
      total: 0,
      next: null,
      operation: null,
    };
    const buttonName = '=';
    if (buttonName === '.') {
      expect(calculate(initialItems, buttonName)).toEqual(
        { total: 0, next: null, operation: null },
      );
    } else if (buttonName === '.') {
      expect(calculate(initialItems, buttonName)).toEqual(
        { total: 0.0, next: null, operation: null },
      );
    } else if (buttonName === '1') {
      expect(calculate(initialItems, buttonName)).toEqual(
        { total: null, next: '1' },
      );
    } else if (buttonName === '=') {
      expect(calculate(initialItems, buttonName)).toEqual({});
    }
  });

  test('operate.js addition test', () => {
    const numberOne = '1';
    const numberTwo = '2';
    const operation = '+';

    expect(operate(numberOne, numberTwo, operation)).toBe('3');
  });

  test('operate.js subtraction test', () => {
    const numberOne = '1';
    const numberTwo = '2';
    const operation = '-';

    expect(operate(numberOne, numberTwo, operation)).toBe('-1');
  });

  test('operate.js multiplication test', () => {
    const numberOne = '1';
    const numberTwo = '2';
    const operation = 'x';

    expect(operate(numberOne, numberTwo, operation)).toBe('2');
  });

  test('operate.js division test', () => {
    const numberOne = '1';
    const numberTwo = '2';
    const operation = 'รท';

    expect(operate(numberOne, numberTwo, operation)).toBe('0.5');
  });

  test('operate.js division by 0 test', () => {
    const numberOne = '1';
    const numberTwo = '0';
    const operation = 'รท';

    expect(operate(numberOne, numberTwo, operation)).toBe('Can\'t divide by 0.');
  });

  test('operate.js remainder test', () => {
    const numberOne = '5';
    const numberTwo = '2';
    const operation = '%';

    expect(operate(numberOne, numberTwo, operation)).toBe('1');
  });

  test('operate.js remainder when divide by 0 test', () => {
    const numberOne = '1';
    const numberTwo = '0';
    const operation = '%';

    expect(operate(numberOne, numberTwo, operation)).toBe('Can\'t find modulo as can\'t divide by 0.');
  });
});

describe('component test', () => {
  it('Calculator component should perform the calculations correctly on initial run', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Quote component should perform correctly on initial run', () => {
    const tree = renderer.create(<Quotes />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('NavBar component should perform correctly on initial run', () => {
    const tree = renderer.create(<Router><NavBar /></Router>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Home component should perform correctly on initial run', () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('Simulation of user clicking button 9 twice', () => {
    render(<Calculator />);
    const buttonEl = screen.getByText(/9/i);
    userEvent.click(buttonEl);
    userEvent.click(buttonEl);
    expect(screen.getByText(/99/i)).toBeInTheDocument();
  });
});
