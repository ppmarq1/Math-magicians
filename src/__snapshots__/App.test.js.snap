// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component test Calculator component should perform the calculations correctly on initial run 1`] = `
<div
  className="mainContainer"
>
  <h2>
    Let's do some Math!
  </h2>
  <div
    className="displayInputs"
  >
    <p
      className="showResult"
    >
      0
       
       
       
    </p>
  </div>
  <div
    className="calcDiv"
  >
    <button
      className="btn"
      onClick={[Function]}
      type="button"
      value="AC"
    >
      AC
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
      value="+/-"
    >
      +/-
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
      value="%"
    >
      %
    </button>
    <button
      className="btn btnclick"
      onClick={[Function]}
      type="button"
      value="รท"
    >
      รท
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
      value="7"
    >
      7
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
      value="8"
    >
      8
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
      value="9"
    >
      9
    </button>
    <button
      className="btn btnclick"
      onClick={[Function]}
      type="button"
      value="x"
    >
      x
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
      value="4"
    >
      4
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
      value="5"
    >
      5
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
      value="6"
    >
      6
    </button>
    <button
      className="btn btnclick"
      onClick={[Function]}
      type="button"
      value="-"
    >
      -
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
      value="1"
    >
      1
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
      value="2"
    >
      2
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
      value="3"
    >
      3
    </button>
    <button
      className="btn btnclick"
      onClick={[Function]}
      type="button"
      value="+"
    >
      +
    </button>
    <button
      className="btnZero"
      onClick={[Function]}
      type="button"
      value="0"
    >
      0
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
      value="."
    >
      .
    </button>
    <button
      className="btn btnclick"
      onClick={[Function]}
      type="button"
      value="="
    >
      =
    </button>
  </div>
</div>
`;

exports[`component test Home component should perform correctly on initial run 1`] = `
<div
  className="homeParagraph"
>
  <h2>
    Welcome to our page!
  </h2>
  <br />
  <br />
  <p
    className="paragraph"
  >
    "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
  </p>
  <br />
  <br />
  <p
    className="paragraph"
  >
    "Lorem ipsum dolor sit amet,Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum."
  </p>
</div>
`;

exports[`component test NavBar component should perform correctly on initial run 1`] = `
<header
  className="nav"
>
  <h1>
    Math Magicians
  </h1>
  <nav
    className="nav-links"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
    <a
      href="/calculator"
      onClick={[Function]}
    >
       Calculator
    </a>
    <a
      href="/quotes"
      onClick={[Function]}
    >
       Quote
    </a>
  </nav>
</header>
`;

exports[`component test Quote component should perform correctly on initial run 1`] = `
<main>
  <p
    className="paragraphQuote"
  >
    Mathematics is not about numbers, equations, computations or algorithms: It is about understanding. - William Paul, Thurston
  </p>
</main>
`;
